plugins/module_utils/network/vyos/config/bgp_address_family/bgp_address_family.py

```diff
@@ -103,6 +132,9 @@ def _compare(self, want, have):
         the `want` and `have` data with the `parsers` defined
         for the Bgp_address_family network resource.
         """
+        if LooseVersion(get_os_version(self._module)) >= LooseVersion("1.4"):
```

Comment:
yeah, but how many ASes are allowed per the router - I thought it is only one; attempts to set it different causes  "Only one bgp instance is allowed per device"  - but I am happy to be told otherwise here - can check with the VyOS dev team

Reply:
Only one, that's the same as 1.4+, which is why I was bringing this up.  However, from the Ansible perspective, you only have one AS in each of the commands too. What I'm wondering is how we handle a new AS under each circumstance.

Here's the big question. Today, the ansible mirrors the experience on the router in 1.3-, you can't change the AS, you must delete the `set protocols bgp XX` in order to send any command (`bgp_global` or `bgp_address_family`) with a different AS.

If we introduce the ability to change the AS in 1.4+ (which more explicitly only has one AS), what do we do if somebody has ansible that contains a `bgp_global` with one AS and a separate `bgp_address_family` with a different AS?  This will cause the AS to flap during configuration. 

This is prevented today in the ansible by the error that happens when you go to send a command with a new AS.

This is a long-winded way of saying that I think we need to consider not allowing configurations in Ansible where the AS number is different between the `bgp_global` (which should be able to change it) and the `bgp_address_family`), which I believe shouldn't be able to change it.  If you use _facts_ in ansible for the AS to keep this consistent across your configuration, you'll never notice it.  However, if you have manual AS numbers in your ansible configurations, that'll cause an error instead of allowing for flapping.

Does that make sense?

Reply:
Did you see https://vyos-community.slack.com/archives/C01A6CJFW1F/p1733380563384319 ? Apparently there is a use case for changing AS (still has to be one per a router). 
Makes sense, we can refuse setting system-as in AF module via argspec, and leave it for global. The doc will need to be cross-referenced, so to save the confusion

Reply:
Added support for AS changed to 1.4 suite
v1.3. remains the same